### ISSUE TITLE
Empty '99-salt.conf' created with ssh-salt test=true

### ISSUE THREAD

**Issue by eapster123** [saltstack/salt#41153]
### Description of Issue/Question
Using **sysctl.present** with salt-ssh and **test=true** creates an empty **/etc/sysctl.d/99-salt.conf** file.  I'm assuming that **test=true** shouldn't create and leave files around.
### Setup
```
# cat common-dev/ipv6.sls 
net.ipv6.conf.all.disable_ipv6:
  sysctl.present:
    - value: 1
net.ipv6.conf.default.disable_ipv6:
  sysctl.present:
    - value: 1
```

### Steps to Reproduce Issue
Check 'tester' before run

```
[tester]# ls -l /etc/sysctl.d
total 0
lrwxrwxrwx 1 root root 14 Apr 13 10:31 99-sysctl.conf -> ../sysctl.conf
```
Run test
```
# salt-ssh -i 'tester' state.apply common-dev/ipv6  test=true
tester:
----------
          ID: net.ipv6.conf.all.disable_ipv6
    Function: sysctl.present
      Result: None
     Comment: Sysctl option net.ipv6.conf.all.disable_ipv6 might be changed, we failed to check config file at /etc/sysctl.d/99-salt.conf. The file is either unreadable, or missing.
     Started: 11:52:45.397433
    Duration: 173.506 ms
     Changes:   
----------
          ID: net.ipv6.conf.default.disable_ipv6
    Function: sysctl.present
      Result: None
     Comment: Sysctl option net.ipv6.conf.default.disable_ipv6 might be changed, we failed to check config file at /etc/sysctl.d/99-salt.conf. The file is either unreadable, or missing.
     Started: 11:52:45.571071
    Duration: 88.491 ms
     Changes:   

Summary for tester
------------
Succeeded: 2 (unchanged=2)
Failed:    0
------------
Total states run:     2
Total run time: 261.997 ms

```
This creates a zero-length file:

```
[tester]# ls -l /etc/sysctl.d
total 0
-rw-r--r-- 1 root root  0 May  9 11:52 99-salt.conf
lrwxrwxrwx 1 root root 14 Apr 13 10:31 99-sysctl.conf -> ../sysctl.conf
```
Using **test=false** works as expected.
### Versions Report
Salt Version:
           Salt: 2016.11.4
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: Not Installed
   pycryptodome: 3.4.3
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 3.10.0-514.16.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core

**Comment by gtmanfred** [comment]
Yup i have replicated this issue.

Also looking at the code it looks like we just create the config file if it doesn't exist when checking what is in the file.

This doesn't just pertain to salt-ssh.

Thanks for reporting, if you wouldn't mind helping fix this we would appreciate it, otherwise I have tagged this as an issue for our platform team.

Thanks,
Daniel

**Comment by lordcirth** [comment]
it seems that the correct behaviour for _check_systemd_salt_config() is simply "return '/etc/sysctl.d/99-salt.conf' and it is only called in one place, default_config() in the same file.  Should I just delete it, and put the string in default_config()?

**Comment by gtmanfred** [comment]
I am actually not truly convinced that this is a real problem.

I think if you wanted to move this, we should leave the logic, and make is
so that it doesn't make the file until after the test=True check.

But we still want to make the file at some point.

**Comment by lordcirth** [comment]
But as my PR mentions, linux_sysctl.persist() creates the file when writing to it, so it's redundant.  And indeed that's the only time you should create it, IMHO, is when you go you write something to it.  A 0-length file doesn't really do anything.

**Comment by lordcirth** [comment]
Just realized I do need to make linux_sysctl.persist() create /etc/sysctl.d, just in case.  I've closed the PR to make a new one against 2016.11.

**Comment by rallytime** [comment]
Fixed with #41255
